superset/reports/commands/execute.py

```diff
@@ -182,11 +193,11 @@ def _get_url(
                 **kwargs,
             )
 
-        # If we need to render dashboard in a specific sate, use stateful permalink
+        # If we need to render dashboard in a specific state, use stateful permalink
         dashboard_state = self._report_schedule.extra.get("dashboard")
         if dashboard_state:
             permalink_key = CreateDashboardPermalinkCommand(
-                dashboard_id=self._report_schedule.dashboard_id,
+                dashboard_id=str(self._report_schedule.dashboard_id),
```

Comment:
Is Mypy not working as expected? Also why should the ID be a string?

Reply:
Mypy actually picked this up. This is due to slugs (`str`) and ids (`int`) being used interchangeably in the codebase (I remember being frustrated about this when working on the permalink feature)